TypeCobol/Compiler/Preprocessor/ReplaceTokensLinesIterator.cs

```diff
@@ -505,6 +508,78 @@ private Token CreateReplacedTokens(Token originalToken, ReplaceOperation replace
             }
         }
 
+        /// <summary>
+        /// Figure out a scan state from the given token.
+        /// </summary>
+        /// <param name="token">The Token to figure out the scan state.</param>
+        /// <returns>The Scan state in any, null otherwise</returns>
+        private static MultilineScanState FigureOutScanState(Token token)
+        {
+            if (token != null)
+            {
+                return token.ScanStateSnapshot ?? (token.TokensLine is TypeCobol.Compiler.Scanner.TokensLine tl
+                            && tl.ScanState != null ? tl.ScanState : null);
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Determines if the given token may be dependent of a state when scanned.
+        /// </summary>
+        /// <param name="token">The token to be checked for scanning state dependency</param>
+        /// <returns>true if the token may be dependent of a scanning state, false otherwise.</returns>
+        private static bool IsScanStateDependent(Token token)
```

Comment:
This could be part of Token class instead of the ReplaceTokensLinesIterator

Reply:
I think Token class is just a model, and does not contain business logic methods.

Reply:
Token class already contains business logic methods. See CompareForReplace, DegradePotentialCodeElementStartingKeywordToSyntaxKeyword, CorrectType.
So we can add one more business method to this class.

Reply:
So maybe this business logic should be included in MultilineScanState which handles the specific logic, it takes the decision of what token type must have a specific treatment.